### DISCUSSION TITLE
Codespaces secrets should get highlighted for visibility

### DISCUSSION THREAD

**Discussion by alfredodeza** [community/community#47463] [Codespaces]
### Select Topic Area

Product Feedback

### Body

The varying levels of exposure, although mentioned, are not clearly highlighted enough. Since these are *secrets* it is easy to overlook the visibility and expose them to users in Codespaces unknowingly. 

There are currently three ways to store Codespaces secrets, each with a varying level of exposure. 

* **Organization-level secret**: Go to an org page, then settings, secrets , then codespaces, for example https://github.com/organizations/$ORG_NAME/settings/secrets/codespaces . That will cause any Codespace in any organization-wide repository to get access to that secret
* **Repository-level secret**: Go to a specific repository, then settings, then  secrets and codespaces. For example: https://github.com/$ORG_NAME/$REPO_NAME/settings/secrets/codespaces . Any person that starts a Codespace from that repository will get access to those secrets.
* **User level settings**: Go to your user settings (click on your user at the top right corner, then settings) and the Codespaces. E.g. https://github.com/settings/codespaces . Any secret for Codespaces created there will be tied to your user only and not available to anyone else.

These constraints are described in each section. For example on this repository:

![Screenshot 2023-02-02 at 9 53 20 AM](https://user-images.githubusercontent.com/317847/218847404-07ee7b13-e5f3-4525-9012-97ecac2b26c2.png)

If a professor gives "Collaborator" access to a student, that means that the student will get those secrets when they create the Codespace.

The ask here is to better highlight the potential disclosure of sensitive information in a Codespace.

**Comment by aishwaryabh** [comment]
Sure I will pass this information on to our team. Thanks for the feedback!

**Comment by alfredodeza** [comment]
@aishwaryabh is there an update on this issue? I think that even a minor update to the wording would be great
